python/cog/director/director.py

```diff
@@ -204,6 +220,12 @@ def _handle_message(self, message_id: str, message_json: str) -> None:
             response=schema.PredictionResponse(**message),
             webhook_caller=webhook_caller(message["webhook"]),
         )
+        self.monitor.set_current_prediction(tracker._response)
+        self._set_span_attributes_from_tracker(span, tracker)
+
+        should_cancel = self.redis_consumer_rotator.get_current().checker(
+            message.get("cancel_key")
+        )
 
         # Override webhook to call us
         message["webhook"] = "http://localhost:4900/webhook"
```

Comment:
In the `try: ... except: ...` blocks below here where we early-return it's _especially_ important to capture the failure (and ideally stuff like exception tracebacks if relevant, or HTTP status codes, etc.). I guess we want to make sure somehow that we set the status `succeeded/canceled/failed` properties and whatever other relevant info we can on every possible exit path from this method.